### ISSUE TITLE
ios: Joining Zoom session doesn't show video or overlay ui, but meeting is connected

### ISSUE THREAD

**Issue by vaughnkoch** [mieszko4/react-native-zoom-us#319]
Hi, I'm seeing unexpected behavior in ios when a user first joins a Zoom session - the user connects and can hear audio, but can't see the Zoom overlay UI. Their camera and mic transmit to the other participants though.

I've checked RNZU issues and the Zoom developer forum but haven't been able to identify the cause. I also checked the changelog of recent Zoom SDK fixes but nothing seems to match. Would it be possible to use Github Discussions for this kind of topic?

## Problem behavior

Once the user connects via RNZU in ios, instead of showing the Zoom overlay UI, it still shows the underlying ZoomScreen react-native components. So there's no visual of the meeting. However, the meeting remains connected. The user can hear the other participants, and the other participants can see and hear the user (so the camera and mic are working fine).

## Expected behavior

When a user loads ZoomScreen in our app, we initiate the Zoom connection via RNZU.
Zoom's Meeting SDK then overlays our app with the fullscreen Zoom UI. The user can see and hear the other participants in the meeting, and they can hear/see the user. This still occurs for the majority of our users.

Notes
- I didn't see anything in app logs that would indicate a problem. I do suppress some logs related to muteMyVideo, although that should govern whether the user's video stream is turned on/off (transmitted to the other participants). In theory, the app should show the UI regardless of whether the video is muted or not.
- This problem is intermittent. It started happening after upgrading RNZU to 6.17.0, which itself upgraded the Zoom SDK to 5.13.10.7064.
- So far, this only occurs when the user first joins the meeting.

Please let me know whether this is a known issue. I would like to narrow this down to RNZU, Zoom SDK or our app code.

Versions:

- RNZoomUs - 6.17.0
- React-Native - 0.71.7
- ZoomSDK - 5.13.10.7064

**Comment by mieszko4** [comment]
Could you try with react-native-zoom-us@6.20.0? It uses 5.14.11.8690 ZoomSDK

**Comment by vaughnkoch** [comment]
I upgraded to react-native-zoom-us@6.20.0 as suggested, but we're still seeing this error where Zoom connects and the mobile app user can be seen and heard on the other end, but the user does not see the ios Zoom overlay.

**Comment by vaughnkoch** [comment]
I think I found the error. I was using MEETING_STATUS_ENDED to detect when the Zoom meeting had ended. However, it appears that Zoom considers the waiting room a meeting as well, so when the user transitions from the waiting room to the actual meeting, Zoom and thus RNZU emits MEETING_STATUS_ENDED.  I confirmed this via testing.

Also see https://devforum.zoom.us/t/getting-meeting-ended-state-when-user-moves-in-meeting-from-waiting-room/41922/2.

Feel free to close this issue.
